src/wazuh_modules/syscollector/tests/sysCollectorImp/syscollectorImp_test.cpp

```diff
@@ -1876,6 +1876,138 @@ TEST_F(SyscollectorImpTest, pushMessageOk1)
+              (R"({"egroup":"root","euser":"root","fgroup":"root","name":"kworker/u256:2-","scan_time":"2020/12/28 21:49:50", "nice":0,"nlwp":1,"pgrp":0,"pid":431625,"ppid":2,"priority":20,"processor":1,"resident":0,"rgroup":"root","ruser":"root","session":0,"sgroup":"root","share":0,"size":0,"start_time":9302261,"state":"I","stime":3,"suser":"root","tgid":431625,"tty":0,"utime":0,"vm_size":0})"_json));
+
+    std::thread t
+    {
+        [&spInfoWrapper]()
+        {
+            Syscollector::instance().init(spInfoWrapper,
+                                          reportFunction,
+                                          reportFunction,
+                                          logFunction,
+                                          SYSCOLLECTOR_DB_PATH,
+                                          "",
+                                          "",
+                                          60, false, true, true, true, true, true, true, true, true, true);
+        }
+    };
+    std::this_thread::sleep_for(std::chrono::seconds{1});
+    Syscollector::instance().push(messageToPush);
+    std::this_thread::sleep_for(std::chrono::seconds{1});
+    Syscollector::instance().destroy();
+
+    if (t.joinable())
+    {
+        t.join();
+    }
+}
+
+TEST_F(SyscollectorImpTest, scanOnDemandOk2)
+{
+    // Only one scan runs, even when sleep time is enough for interval to expire
+    constexpr auto messageToPush{R"(syscollector_run)"};
+    const auto spInfoWrapper{std::make_shared<SysInfoWrapper>()};
+    EXPECT_CALL(*spInfoWrapper, hardware())
+    .Times(1)
+    .WillRepeatedly(Return(nlohmann::json::parse(
+                               R"({"board_serial":"Intel Corporation","scan_time":"2020/12/28 21:49:50", "cpu_MHz":2904,"cpu_cores":2,"cpu_name":"Intel(R) Core(TM) i5-9400 CPU @ 2.90GHz","ram_free":2257872,"ram_total":4972208,"ram_usage":54})")));
+
+    EXPECT_CALL(*spInfoWrapper, os()).Times(0);
+    EXPECT_CALL(*spInfoWrapper, packages(_)).Times(0);
+    EXPECT_CALL(*spInfoWrapper, networks()).Times(0);
+    EXPECT_CALL(*spInfoWrapper, processes(_)).Times(0);
+    EXPECT_CALL(*spInfoWrapper, ports()).Times(0);
+    EXPECT_CALL(*spInfoWrapper, hotfixes()).Times(0);
+
+    std::thread t
+    {
+        [&spInfoWrapper]()
+        {
+            Syscollector::instance().init(spInfoWrapper,
+                                          reportFunction,
+                                          reportFunction,
+                                          logFunction,
+                                          SYSCOLLECTOR_DB_PATH,
+                                          "",
+                                          "",
+                                          5, false, true, false, false, false, false, false, false, false, true);
+        }
+    };
+    std::this_thread::sleep_for(std::chrono::seconds{3});
+    Syscollector::instance().push(messageToPush);
+    std::this_thread::sleep_for(std::chrono::seconds{4});
```

Comment:
If I'm not making a bad calculus, this leaves only 1 second before having a successful interval scan. Maybe we are at the edge of time to have a flaky test. I would consider sleeping for three seconds again.

Reply:
It's a good precaution. Fixed!